File: products/LoanHub/apps/frontend/types/branch.ts
export type Branch = {
    id: string;
    company_id: string;

    name: string;
    district: string;
    town: string | null;
    address: string | null;

    phone: string | null;
    email: string | null;

    is_active: boolean;
    is_headquarters: boolean;

    created_at?: string;
    updated_at?: string;
};

export type BranchCreatePayload = {
    company_id: string;
    name: string;
    district: string;
    town?: string | null;
    address?: string | null;
    phone?: string | null;
    email?: string | null;
    is_active?: boolean;
    is_headquarters?: boolean;
};

export type BranchUpdatePayload = Partial<
    BranchCreatePayload
>;

File: products/LoanHub/apps/backend/database/models/branch.py
from sqlalchemy import Boolean, Column, ForeignKey, String, Text, UniqueConstraint
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from database.base import Base


class CompanyBranch(Base):
    __tablename__ = "company_branches"
    __table_args__ = (
        UniqueConstraint(
            "company_id",
            "name",
            name="uq_company_branch_name",
        ),
    )

    company_id = Column(
        UUID(as_uuid=True),
        ForeignKey("loan_companies.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )

    name = Column(String(150), nullable=False)
    district = Column(String(100), nullable=False)
    town = Column(String(100), nullable=True)
    address = Column(Text, nullable=True)

    phone = Column(String(30), nullable=True)
    email = Column(String(150), nullable=True)
    is_active = Column(Boolean, nullable=False, default=True)
    is_headquarters = Column(Boolean, nullable=False, default=False, index=True)

    company = relationship("LoanCompany", back_populates="branches")
    staff_members = relationship("CompanyStaff", back_populates="branch")
    loans = relationship("ClientCompanyLoan", back_populates="branch")
